adapters/adhese/adhese.go

```diff
@@ -0,0 +1,282 @@
+package adhese
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type AdheseAdapter struct {
+	http             *adapters.HTTPAdapter
+	endpointTemplate template.Template
+	dummyCacheBuster int
+}
+
+type AdheseKeywordsParams struct {
+	Key    string   `json:"key,omitempty"`
+	Values []string `json:"value,omitempty"`
+}
+
+func (a *AdheseAdapter) Name() string {
+	return "adhese"
+}
+
+func (a *AdheseAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func extractSlotParameter(parameters openrtb_ext.ExtImpAdhese) string {
+	return fmt.Sprintf("/sl%s-%s", parameters.Location, parameters.Format)
+}
+
+func extractTargetParameters(parameters openrtb_ext.ExtImpAdhese) string {
+	if parameters.Keywords == nil || len(parameters.Keywords) == 0 {
+		return ""
+	}
+	m := make(map[string][]string)
+	for _, kv := range parameters.Keywords {
+		for _, tv := range kv.Values {
+			cur, _ := m[kv.Key]
+			new := cur[:]
+			m[kv.Key] = append(new, tv)
+		}
+	}
+
+	var parametersAsString = ""
+
+	for k, v := range m {
+		parametersAsString += "/" + k + strings.Join(v, ";")
+	}
+	params := strings.Split(parametersAsString, "/")
+	sort.Strings(params)
+	return strings.Join(params, "/")
+
+}
+
+func extractGdprParameter() string {
+	//const gdprParams = (gdprConsent && gdprConsent.consentString) ? [ 'xt' + gdprConsent.consentString, 'tlall' ] : [];
```

Comment:
How about that, and the one below? Not supported by prebid-server or any other reason?

Reply:
I have no idea, that's why I left it like this
The docs aren't so helpful: https://github.com/prebid/prebid-server/blob/master/docs/developers/add-new-bidder.md

Reply:
The consent string (/xt) should be available like in `extUser.Consent` field, see:
https://github.com/prebid/prebid-server/blob/f1f2b094a9c202300087a3c9bcc99c06e711e9c6/adapters/adform/adform.go#L473

referer: `request.Site.Page`, see
https://github.com/prebid/prebid-server/blob/f1f2b094a9c202300087a3c9bcc99c06e711e9c6/adapters/adform/adform.go#L447